conv.go

```diff
@@ -11,27 +12,30 @@ import (
 // Note it may break if string and/or slice header will change
 // in the future go versions.
 func B2S(b []byte) string {
```

Comment:
I think this code could be simplified by `return *(*string)(unsafe.Pointer(&b))`

Reply:
Because the size of a byte slice is larger than a string, and their internal structures are similar, so the conversion is valid.

Reply:
Yes, that works as long as the Go developers keep the internal structures synced. The `reflect` documentation explicitly states that one should not rely on this to be the case, but I think it might be a reasonable assumption still. As long as you don't explicitly depend on the layout, but just depend on that `reflect.StringHeader` and `reflect.SliceHeader` stay similar, I think it's okay.

It would only fail if Go were to change the order of `Len` and `Data` in only one of the internal header structures, which I think is pretty unlikely.

Reply:
Yeah, that's it.